bin/picom-trans

```diff
@@ -60,74 +60,87 @@ print_usage()
+          echo "$EXE_NAME: option ${1%=} needs a value" >&2
+          exit 1;;
+        (--opacity|--name|--window)
+          test $i -eq $# \
+            && echo "$EXE_NAME: option $1 needs a value" >&2 \
+            && exit 1;;
+      esac
+
+      # Separate  "--ARG=VAL"  into  "--ARG" "VAL"
+      case "$1" in
+        (--opacity=*|--name=*|--window=*)
+          ARG="$(echo "$1" | sed -E 's/(--[^=]+)=.*$/\1/')"
+          VAL="${1##${ARG}=}"
+          shift && set -- "$ARG" "$VAL" "$@"
+      esac
+
+      # Turn into short form
+      case "$1" in
+        (--help|--opacity|--get|--delete|--toggle|--reset|--select|--current|--name|--window)
+          ARG=${1#-}                          # remove one '-' from prefix
+          ARG="$(echo "$ARG" | cut -c -2)"    # get first two characters
+          shift && set -- "$ARG" "$@"
+      esac
+
+      # If the argument still starts with --, it is an invalid argument
+      case "$1" in
+        (--*)
+          echo "$EXE_NAME: illegal option $1" >&2
+          exit 1
+      esac
+    fi
+    #### [END] Convert GNU longopts to POSIX equivalents ####
+
+
+    #### [START] Prepend '-o' to  standalone opacity values ####
+    # Iterate over every argument and check if it is an opacity without the -o
+    # option in the previous argument. If so, then prepend the -o option.
+    # e.g. Turn this -
+    #     picom-trans -c +10 -s
+    # into this -
+    #     picom-trans -c -o +10 -s
+    #
+    # NOTE: Don't touch arguments that are preceded by -n or -w
+    #       i.e. don't touch arguments that take a value.
+    # i.e. This -
+    #     picom-trans -w 75 -o 90
+    # should NOT be turned into this -
+    #     picom-trans -w -o 75 -o 90
     if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$' && \
-      ! eval "echo \${$#}" | grep -q '^-[hdtrgsnwc]*o$'
-    # NOTE: eval "echo \${$#}" means 'echo the last argument'
-    # NOTE: The letters inside the square brackets (ie. hdtrgsnwc) are
-    #       the same as those in the getopts argument, but without the 'o'
+      ! eval echo "\"\${$#}\"" | grep -q '^-[hdtrgsc]*[own]$'
+    # NOTE: eval echo \${$#} means 'echo the last argument'
+    # NOTE: The letters inside the first  square brackets (ie. hdtrgsc) are
+    #       the same as those in the getopts argument, minus those that are
+    #       followed by a ':'
+    # NOTE: The letters inside the second square brackets (ie. own)     are
+    #       the same as those in the getopts argument, minus those that are NOT
+    #       followed by a ':'
     then
       OPACITY=$1
-      shift && set -- "-o" "$OPACITY" "$@"
+      shift && set -- "$@" "-o" "$OPACITY"
+      i=$(( i + 2 ))
+    else
+      ARG="$1"
+      shift && set -- "$@" "$ARG"
+      i=$(( i + 1 ))
     fi
+    #### [END] Prepend '-o' to  standalone opacity values ####
+
 
     # Prepare for next iteration
-    FIRST="$1"
-    shift && set -- "$@" "$FIRST"
+    ARG="$1"
+    shift && set -- "$@" "$ARG"
     i=$(( i + 1 ))
```

Comment:
We are now shifting to many arguments around. Either drop the `else` branch and `continue` after fixing opacity, or move *Prepare for next iteration* into the `else` branch. Or only add `-o` to fix opacity (and increment `i` to accommodate for the extra argument), drop the `else` branch and handle them normally in *Prepare for next iteration*.
I personally would prefer the first or last option.

Reply:
Please test if 8e9f9d9 fixes it.

I am unable to test right now due to booting issues. Will try testing on OpenBSD day-after-tomorrow.

Reply:
Should work as expected now. At least does so on my system.

Reply:
Checked on OpenBSD. Works.